src/cache.c

```diff
@@ -91,9 +99,14 @@ struct cache_entry *dllist_remove_tail(struct cache *cache)
  */
 struct cache *cache_create(int max_size, int hashsize)
 {
-    ///////////////////
-    // IMPLEMENT ME! //
-    ///////////////////
+    struct cache *c = malloc(sizeof(struct cache));
+    (void)hashsize;
+    c->index = hashtable_create(max_size, NULL);
+    c->max_size = max_size;
+    c->cur_size = 0;
+    c->head = NULL;
```

Comment:
c->head=cache->tail=NULL, just another way to write this cleaner? Not necessarily more readable thoigh